File: ch003_stack/stack.py
from typing import Optional, Any

class Stack:
  def __init__(self, arr: Optional[list[Any]]=None):
    if arr is None:
      self.arr = []
    else:
      self.arr = arr
  
  def append(self, val: Any):
    self.arr.append(val)

  def pop(self) -> Any:
    return self.arr.pop()
  
  def peek(self)-> Any:
    return self.arr[-1]
  
  def __str__(self) -> str:
    return f'[{",".join(str(x) for x in self.arr)}]'
  
s = Stack(list(range(10)))
s.append(101)
print(s)
s.pop()
print(s)
print(s.peek())

File: ch003_stack/stack.js
class Stack {
  constructor(arr = []) {
    this.arr = arr;
  }

  push(value) {
    this.arr.push(value);
  }

  pop() {
    return this.arr.pop();
  }

  peek() {
    return this.arr[this.arr.length - 1];
  }
}

l = [1, 2, 3, 4, 5];
stack = new Stack(l);

stack.push(101);
console.log(stack, l);
x = stack.pop();
console.log(stack, x);
console.log(stack.peek());
